code/modules/antagonists/cult/cult_items.dm

```diff
@@ -625,20 +625,31 @@
 	sharpness = IS_SHARP
 	hitsound = 'sound/weapons/bladeslice.ogg'
 	var/datum/action/innate/cult/spear/spear_act
+	var/wielded = FALSE // track wielded status on item
 
-/obj/item/twohanded/cult_spear/Initialize()
+/obj/item/cult_spear/ComponentInitialize()
 	. = ..()
 	AddComponent(/datum/component/butchering, 100, 90)
+	AddComponent(/datum/component/two_handed, force_unwielded=17, force_wielded=24, \
+				on_wield_callback=CALLBACK(src, .proc/on_wield), on_unwield_callback=CALLBACK(src, .proc/on_unwield))
 
-/obj/item/twohanded/cult_spear/Destroy()
+/// Callback triggered on wield of two handed item
+/obj/item/cult_spear/proc/on_wield(mob/user)
+	wielded = TRUE
+
+/// Callback triggered on unwield of two handed item
+/obj/item/cult_spear/proc/on_unwield(mob/user)
+	wielded = FALSE
+
+/obj/item/cult_spear/update_icon_state()
```

Comment:
again, this behavior should be generalized by having an 1h_icon (or using intial icon) and 2h_icon var on the component and listening for update_icon

Reply:
cult_spear uses wielded in the hit reaction logic so this is needed.

Reply:
now managed by icon_prefix